bundles/org.openhab.core.karaf/src/main/java/org/openhab/core/karaf/internal/FeatureInstaller.java

```diff
@@ -446,24 +478,26 @@ private static synchronized boolean uninstallFeature(FeaturesService featuresSer
         return false;
     }
 
-    private static synchronized boolean installPackage(FeaturesService featuresService,
-            final Map<String, Object> config) {
+    private synchronized boolean installPackage(FeaturesService featuresService, final Map<String, Object> config) {
         boolean configChanged = false;
         Object packageName = config.get(OpenHAB.CFG_PACKAGE);
         if (packageName instanceof String) {
             currentPackage = (String) packageName;
-            String name = PREFIX + PREFIX_PACKAGE + ((String) packageName).trim();
-            if (installFeature(featuresService, name)) {
-                configChanged = true;
+            String fullName = PREFIX + PREFIX_PACKAGE + ((String) packageName).trim();
+            if (currentPackage.equals(MINIMAL_PACKAGE)) {
```

Comment:
Position literals first in `String` comparisons - that way if the `String` is `null` you won't get a `NullPointerException`, it'll just return `false`. [source: PMD](http://pmd.sourceforge.net/pmd-4.3.0/rules/design.html#PositionLiteralsFirstInComparisons)

Reply:
We are inside an "instanceof String" check, so we can be sure it is not null.

Reply:
But is still a good habit 😜